pkg/scheduler/framework/plugins/noderesources/balanced_allocation_test.go

```diff
@@ -351,7 +350,9 @@ func TestNodeResourcesBalancedAllocation(t *testing.T) {
 	for _, test := range tests {
 		t.Run(test.name, func(t *testing.T) {
 			snapshot := cache.NewSnapshot(test.pods, test.nodes)
-			fh, _ := runtime.NewFramework(nil, nil, wait.NeverStop, runtime.WithSnapshotSharedLister(snapshot))
+			stopCh := make(chan struct{})
+			defer close(stopCh)
+			fh, _ := runtime.NewFramework(nil, nil, stopCh, runtime.WithSnapshotSharedLister(snapshot))
```

Comment:
Could you use `context.WithCancel(context.Background())` and `ctx.Done()` like other changes?
so that you can use that `ctx` in L358 `p.(framework.ScorePlugin).Score(context.Background(), nil, test.pod, test.nodes[i].Name)`